### ISSUE TITLE
ST Table no longer imports

### ISSUE THREAD

**Issue by daedalus1776** [garysharp/EduHub.Data#3]
There is an error when trying to run $eduhub.ST.Importtosql, but only with this table. All other Eduhub data appears to work fine

**Comment by daedalus1776** [comment]
Sorry, I should have mentioned this is in powershell that I've experienced the issue

**Comment by garysharp** [comment]
Hi @daedalus1776 - any chance you could share what that error is? I'm not seeing it myself.

**Comment by daedalus1776** [comment]
Thanks for the response.
I run the following command:
`$eduHub.ST.WriteToSqlServer($ipaddy,"eduHub",$eduhubSQLU,$eduhubSQLP);`

And receive the following error:
```
Exception calling "WriteToSqlServer" with "4" argument(s): "One or more errors 
occurred."
At C:\Program Files (x86)\eduHub\eduHub-curric-sync.ps1:6 char:1
+ $eduHub.ST.WriteToSqlServer($ipaddy,"eduHub",$eduhubSQLU,$eduhubSQLP);
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : AggregateException
```


the SF table, for instance, imports with the same arguments without issue.

Thanks!

**Comment by garysharp** [comment]
Unfortunately that doesn't provide us with much information.
Can you investigate the `$Error` variable and dig into the aggregate exceptions (typically via `InnerExceptions`). For example:
- `$Error[0]`
- `$Error[0].Exception`
- `$Error[0].Exception.InnerExceptions`
- `$Error[0].InnerExceptions`

**Comment by daedalus1776** [comment]
It doesn't give me much:

```
PS C:\Program Files (x86)\eduHub> $error[0]
Exception calling "WriteToSqlServer" with "4" argument(s): "One or more errors 
occurred."
At line:1 char:1
+ $eduHub.ST.WriteToSqlServer($ipaddy,"eduHub",$eduhubSQLU,$eduhubSQLP);
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : AggregateException
 

PS C:\Program Files (x86)\eduHub> $error[0].Exception
Exception calling "WriteToSqlServer" with "4" argument(s): "One or more errors 
occurred."

PS C:\Program Files (x86)\eduHub> $error[0].Exception.InnerException
One or more errors occurred.
```

The most I could get was this:

```
PS C:\Program Files (x86)\eduHub> $Error[0].Exception.StackTrace
   at System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationExce
ption(Exception exception, Type typeToThrow, String methodName, Int32 numArgs, Member
Info memberInfo)
   at CallSite.Target(Closure , CallSite , Object , String , String , Object , Object
 )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute5[T0,T1,T2,T3,T4,TRet](CallSite 
site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4)
   at System.Management.Automation.Interpreter.DynamicInstruction`6.Run(InterpretedFr
ame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(In
terpretedFrame frame)
```

I hope that's at all helpful

**Comment by garysharp** [comment]
What about the `.InnerExceptions` property @daedalus1776 ? And possibly check `$Error[1]` also (and maybe `$Error[2]`?)

**Comment by daedalus1776** [comment]
Hey @garysharp, sorry that doesn't really give anything:

```
PS C:\Program Files (x86)\eduHub> $Error[0]
Exception calling "WriteToSqlServer" with "4" argument(s): "One or more errors 
occurred."
At C:\Program Files (x86)\eduHub\eduHub-curric-sync.ps1:6 char:1
+ $eduHub.ST.WriteToSqlServer($ipaddy,"eduHub",$eduhubSQLU,$eduhubSQLP);
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : AggregateException
 

PS C:\Program Files (x86)\eduHub> $Error[1]

PS C:\Program Files (x86)\eduHub> $Error[0].InnerExceptions

PS C:\Program Files (x86)\eduHub> 
```

Could it be caused by the ST table itself having something erroneous contained? I could try splitting the file in half and trying both halves to see if that results in any change

**Comment by garysharp** [comment]
@daedalus1776 can you please reach out to me via Skype for Business (edumail) or Slack (DET IT Division)

**Comment by garysharp** [comment]
Thanks for the chat @daedalus1776.

The error was eventually resolved at: `$Error[0].Exception.InnerException.InnerExceptions` 🙃

The eduHub data schema has changed since the current release. In particular (at least) one column size has been increased. This causes an issue if the increased limit is used by the csv files.

I'll reach out to contacts for a newer version of the schema. In the interim we manually increased the column size which resolved the issue.

**Comment by garysharp** [comment]
@daedalus1776 this should be resolved using [Version 1.64.19](https://github.com/garysharp/EduHub.Data/releases/tag/v1.64.19).
